### PR TITLE
Closes #936 funnel step reordering

### LINKED ISSUE #936: Support funnel step reordering
**Is your feature request related to a problem? Please describe.**
I need to add a new step at the start of my funnel but I can't just insert a new one or move one to the start. I have to delete all existing actions+filters and start again.

**Describe the solution you'd like**
To click + drag actions to reorder them in the list.

**Describe alternatives you've considered**

**Additional context**

***Thank*** you for your feature request - we love each and every one!

### PR DESCRIPTION
**Pull request by glmaljkovich**
drag2
drag

closes #936 

## Changes

- Allow drag-and-drop reordering of Funnel steps

## Checklist

- [ ] All querysets/queries filter by Team (if applicable)
- [ ] Backend tests (if applicable)
- [ ] Cypress E2E tests (if applicable)

### CODE REVIEW

File: frontend/src/scenes/trends/ActionFilter/ActionFilter.js [hunk 1]
```diff
@@ -1,33 +1,77 @@
...
-                        logic={logic}
-                        filter={filter}
-                        index={index}
-                        key={index}
-                        hideMathSelector={hideMathSelector}
-                    />
-                ))}
+        <div style={{ position: 'relative' }}>
+            {localFilters && (
+                <ReactGridLayout
```
Comment (timgl) [inline frontend/src/scenes/trends/ActionFilter/ActionFilter.js:56]: The problem is that this component is also used in /trends, which doesn't need to be re-ordered but in this case is also crashing

**Review by timgl — COMMENTED** [review]
Awesome work! There is a bit of a bug with /trends because of this

File: frontend/src/scenes/trends/ActionFilter/ActionFilter.js (hunk 1 above)
Comment (timgl) [reply frontend/src/scenes/trends/ActionFilter/ActionFilter.js:56]: Just thinking about it, I think it's a good thing to be able to re-order the actions on /trends too, the only thing we need to do is make sure it doesn't crash the trends page :)

Comment (glmaljkovich) [reply frontend/src/scenes/trends/ActionFilter/ActionFilter.js:56]: Oops. Didn't notice that 😅 . Let me fix it

Comment (glmaljkovich) [reply frontend/src/scenes/trends/ActionFilter/ActionFilter.js:59]: It should be fixed now. I was able to add and reorder filters at ease in both pages.
add_filter

**Review by timgl — COMMENTED** [review]
Almost there! Two quick things:
1. The remove button now overflows out of the sidebar on trends
image

2. I prefer `<MenuOutlined>` as the drag icon over the current one. It could also be a bit smaller

**Comment by glmaljkovich** [comment]
Now on the Trends page the drag handle is outside the container.
outside 2
outside

The Funnel page keeps the previous design
inside

**Comment by timgl** [comment]
@glmaljkovich Sorry for the late response, i completely missed this!

I do have another issue that I tried to solve (but failed): you can't add two events of the same name anymore (so two `$pageview` events for example). I think it's because you're using the id, but they aren't necessarily unique. I tried solving this using the index but I don't think that's the right approach here. Do you have any thoughts?

**Comment by glmaljkovich** [comment]
Hey, sorry. I completely lost track of this. I added a new property `layoutId` that uses `id + order`. Seems to work fine.
That being said, the fact that React Grid Layout doesn't detect content resize is a true nightmare. Here's an example of an unhandled overflow:

overflow2

**Comment by paolodamico** [comment]
+1 for this feature, hope we can merge it soon

**Pull request closed without merging**